git/libgit2/repository_client.go

```diff
@@ -249,21 +249,16 @@ func (l *Client) Commit(info git.Commit, signer *openpgp.Entity) (string, error)
 	defer index.Free()
 
 	// add to index any files that are not within .git/
-	if err = filepath.Walk(l.repository.Workdir(),
+	if err = util.Walk(l.repoFS, "",
 		func(path string, info os.FileInfo, err error) error {
 			if err != nil {
 				return err
 			}
-
-			rel, err := filepath.Rel(l.repository.Workdir(), path)
-			if err != nil {
-				return err
-			}
-			if info.IsDir() || strings.HasPrefix(rel, ".git") || rel == "." {
+			if info.IsDir() || strings.HasPrefix(path, ".git") || path == "" {
```

Comment:
Previous code checked for `"."` instead of `""`, is this just due to the underlying library only? Is there any impact?

Reply:
It's because we pass `""` as the `root` path to `util.Walk()`. `util.Walk()` calls the callback recursively on each file presented inside the chrooted filesystem, but on the first iteration, it calls the callback on the `root` path, which is `""` in our case. (ref: https://github.com/go-git/go-billy/blob/master/util/walk.go#L18)